ta/mk/ta_dev_kit.mk

```diff
@@ -65,9 +76,24 @@ clean:
 
 subdirs = .
 include  $(ta-dev-kit-dir)/mk/subdir.mk
+
+#the build target is ta
+ifneq ($(binary),)
 vpath %.c $(ta-dev-kit-dir)/src
 srcs += user_ta_header.c
+endif
 
 include  $(ta-dev-kit-dir)/mk/gcc.mk
 include  $(ta-dev-kit-dir)/mk/compile.mk
+ifneq ($(binary),)
 include  $(ta-dev-kit-dir)/mk/link.mk
+else
+ifneq ($(libname),)
+all: $(libname).a
+cleanfiles += $(libname).a
+
+$(libname).a: $(objs)
+	@echo '  AR      $(libname)'
+	$(q)rm -f $(libname) && $(AR$(sm)) rcs -o $(libname).a $^
```

Comment:
Since it seems `.a` isn't supposed to be included in `$(libname)` please update the commit message to reflect that also.

Please replace `$(libname)` and `$(libname).a` with `$@` instead, the same on the line above.